lib/ASTGen/Sources/ASTGen/SourceManager+MacroExpansionContext.swift

```diff
@@ -43,7 +43,7 @@ extension String {
   /// Retrieve the base name of a string that represents a path, removing the
   /// directory.
   var basename: String {
-    guard let lastSlash = lastIndex(of: "/") else {
+    guard let lastSlash = lastIndex(where: { ["/", "\\"].contains($0) }) else {
```

Comment:
This condition should be specialized for Windows specifically

Reply:
Avoiding the special casing is nice - is there some case that you expect this to be problematic for on other platforms?

Reply:
Special cased Apple and Linux platforms instead.